modules/core/src/v2/coins/utxo/recovery/baseApi.ts

```diff
@@ -0,0 +1,86 @@
+/**
+ * @prettier
+ */
+import * as superagent from 'superagent';
+import { BitGoJsError } from '../../../../errors';
+
+export class ApiNotImplementedError extends BitGoJsError {
+  constructor(coinName: string) {
+    super(`api not implemented for coin ${coinName}`);
+  }
+}
+
+export class ApiRequestError extends BitGoJsError {
+  constructor(url: string, reason: Error | string) {
+    super(`error in API request: ${url}: ${reason}`);
+  }
+
+  static forRequest(req: superagent.Request, reason: Error | string): ApiRequestError {
+    return new ApiRequestError(req.toString(), reason);
+  }
+
+  static forResponse(res: superagent.Response, reason: Error | string): ApiRequestError {
+    return new ApiRequestError(res.header, reason);
+  }
+}
+
+export type RequestOptions = {
+  retry?: number;
+};
+
+export type Response<T> = {
+  map<V>(f: (body: T) => V): V;
+};
+
+export class BaseApi {
+  constructor(public baseUrl: string) {
+    if (!baseUrl.startsWith('https://')) {
+      throw new Error(`baseUrl must start with https://`);
+    }
+    if (baseUrl.endsWith(`/`)) {
+      throw new Error(`baseUrl must not end with slash (/)`);
+    }
+  }
+
+  async request<T>(
+    method: string,
+    path: string,
+    requestBody: unknown | undefined,
+    params: RequestOptions = {}
+  ): Promise<Response<T>> {
+    if (!path.startsWith(path)) {
+      throw new Error(`path must start with '/' character`);
+    }
+    const url = this.baseUrl + path;
+    let response;
+    try {
+      response = await superagent(method, url)
+        .retry(params.retry)
+        // eslint-disable-next-line @typescript-eslint/ban-types
+        .send(requestBody as object);
```

Comment:
can you use `Record` here to avoid the `ban-types` violation?

Reply:
indeed I can, will add it to the follow-up PR